Add wip Haskell example

t4ccer opened the pull request:
I've been experimenting with dream2nix recently and implemented a simple haskell support via cabal freeze and build plans. Far from being production ready but _maybe_ could be used as a base

Known issues/disclaimers:
- ~https://github.com/ekmett/free doesn't build because some `sdist` magic is going on and `/doc` dir is moved~
- No automatic detection of required c libraries
- Not tested with alternative package registries like https://input-output-hk.github.io/cardano-haskell-packages/
- Tests are not ran
- Dependencies and main projects are built in one derivation so no caching
- Not tested how much % of hackage builds

DavHau commented on examples/packages/single-language/haskell-cabal/README.md:
````diff
@@ -0,0 +1,13 @@
+## Generate lock file
+
+```console
+$ nix run .#checks.x86_64-linux.example-package-haskell-cabal.lock
````
I really appreciate the effort to add some docs, but these attribute paths are not portable.
Once the example gets copied to another flake, these attribute paths are not guaranteed to be the same.

Maybe remove this file and extend the `./docs` instead by adding a guide (can be separate PR)?

DavHau reviewed (changes requested):
Very nice work. Would you like to maintain this module in the future?

EDIT: Oh, and sorry for the late review, I've been a bit busy. Will try to be more responsive.

t4ccer commented:
Hey, thanks for having a look. I can maintain the Haskell module in my spare time, sure, although I don't have time to actively work on improvements right now, though I'll try addressing some issues in the future
